packages/eslint-plugin/lib/rules/no-react-navigation-stack.ts

```diff
@@ -0,0 +1,50 @@
+import type { Rule } from "eslint";
+
+// Custom Rule: No Import from @react-navigation/stack
+export const noReactNavigationStackImportRule: Rule.RuleModule = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Disallow importing from `@react-navigation/stack` and suggest using `@react-navigation/native-stack` instead.",
+      category: "Best Practices",
+      recommended: true,
+      url: "https://github.com/bamlab/react-native-project-config/tree/main/packages/eslint-plugin/docs/rules/no-react-navigation-stack.md",
+    },
+    messages: {
+      noReactNavigationStackImport:
+        'Please use "@react-navigation/native-stack" instead of "@react-navigation/stack".',
```

Comment:
"@react-navigation/native-stack""
```suggestion
        '"@react-navigation/native-stack" provides out of the box native screens and native transitions for better performance and user experience.',
```